### PR TITLE
Blanc pull request

### PR DESCRIPTION
**Pull request by EvgenyKos**
Nina, please review the pull request and I will merge it with bert

### CODE REVIEW

File: blanc/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# BLANC extention into machine translation quality estimation
```
Comment (ninalopatina) [inline blanc/README.md]: "extension to"

File: blanc/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+This is an extension of BLANC-help into machine translation quality estimation. The original BLANC version is applied on evaluating the quality of text summarization. I used the original sentence  (Russian language) as a document and translation (English language) as a summary to calculate the relative score. Our assumption was that a model should predict more accurately with the translation rather than with a filler. To get the scores I tested 3 models: bert-base-multilingual-uncased, bert-base-multilingual-cased and further pretrained Bert base uncased model. Further pretrained Bert model did not improve the results beyond the base model. It was tested on multiple pretrained Bert models with different hyperparameters. At the same time, the "cased" base model predicted 12 times more [UNK] tokens. During the initial testing it was uncovered that Bert was masking [UNK] tokens. It does not give practical reasons to predict words that are unknown in masked sentences. Therefore, I removed [UNK] tokens from masking.
```
Comment (ninalopatina) [inline blanc/README.md]: the reference for our expectation that the prediction should be predicted more accurately with the translation should be linked via markdown

File: blanc/README.md (hunk 1 above)
Comment (ninalopatina) [inline blanc/README.md]: this directory should be named blanc_mt to be more clear

File: blanc/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Instructions
+
+1. Download the data https://www.quest.dcs.shef.ac.uk/wmt20_files_qe/ru-en.tar.gz and extract into directory.
+
+2. Calculate the relative score by executing `python run.py`.
+   
+   I used the following specification to achive the hisgest score:
```
Comment (ninalopatina) [inline blanc/README.md]: highest

File: blanc/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+This is an extension of BLANC-help into machine translation quality estimation. The original BLANC version is applied on evaluating the quality of text summarization. I used the original sentence  (Russian language) as a document and translation (English language) as a summary to calculate the relative score. Our assumption was that a model should predict more accurately with the translation rather than with a filler. To get the scores I tested 3 models: bert-base-multilingual-uncased, bert-base-multilingual-cased and further pretrained Bert base uncased model. Further pretrained Bert model did not improve the results beyond the base model. It was tested on multiple pretrained Bert models with different hyperparameters. At the same time, the "cased" base model predicted 12 times more [UNK] tokens. During the initial testing it was uncovered that Bert was masking [UNK] tokens. It does not give practical reasons to predict words that are unknown in masked sentences. Therefore, I removed [UNK] tokens from masking.    
+
+## Instructions
+
+1. Download the data https://www.quest.dcs.shef.ac.uk/wmt20_files_qe/ru-en.tar.gz and extract into directory.
```
Comment (ninalopatina) [inline blanc/README.md]: into which dir? it should be a data dir in the blanc dir, or the data dir in the root dir

File: blanc/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# BLANC extention into machine translation quality estimation
+
+This is an extension of BLANC-help into machine translation quality estimation. The original BLANC version is applied on evaluating the quality of text summarization. I used the original sentence  (Russian language) as a document and translation (English language) as a summary to calculate the relative score. Our assumption was that a model should predict more accurately with the translation rather than with a filler. To get the scores I tested 3 models: bert-base-multilingual-uncased, bert-base-multilingual-cased and further pretrained Bert base uncased model. Further pretrained Bert model did not improve the results beyond the base model. It was tested on multiple pretrained Bert models with different hyperparameters. At the same time, the "cased" base model predicted 12 times more [UNK] tokens. During the initial testing it was uncovered that Bert was masking [UNK] tokens. It does not give practical reasons to predict words that are unknown in masked sentences. Therefore, I removed [UNK] tokens from masking.    
+
+## Instructions
+
```
Comment (ninalopatina) [inline blanc/README.md]: missing step to install req.txt; also req file is missing pandas

File: blanc/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+1. Download the data https://www.quest.dcs.shef.ac.uk/wmt20_files_qe/ru-en.tar.gz and extract into directory.
+
+2. Calculate the relative score by executing `python run.py`.
+   
+   I used the following specification to achive the hisgest score:
+   
+   model_name = 'bert-base-multilingual-uncased',
+   
+   gap = 2,
```
Comment (ninalopatina) [inline blanc/README.md]: are these the defaults? if so, this should be indicated. if not, they should be

File: blanc/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+   
+   min_token_length_normal = 4, 
+   
+   min_token_length_lead = 5. 
+   
+   To get the scores on the randomaly shuffled translations sentences and translations with randomly replaced words specify the 'type' (view the documentation below). 
+
+3. Evaluate the results using notebook `blanc_data_analysis.ipynb`.
```
Comment (ninalopatina) [inline blanc/README.md]: this should be in a notebooks dir, not results

File: blanc/bert-further-training/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# BERT-further-pretrainig
```
Comment (ninalopatina) [inline blanc/bert-further-training/README.md]: see feedback from your other repo

File: blanc/bert-further-training/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+## Instructions
+
+1. Download and extract corpuses into directory. Create new enviroment and install requirements.txt
+
+   1.1. pip install -r requirements.txt
+
+2. Training
+
+   2.1. Specicy GPU (default 0). For CPU type --GPU -1
```
Comment (ninalopatina) [inline blanc/bert-further-training/README.md]: the format for indicating the possible arguments, please use the same format as your other readme

File: blanc/README.md (hunk 7 above)
Comment (ninalopatina) [reply blanc/README.md]: Also, the path in the notebook is content/base.....csv; but you have those files in the directory

File: blanc/results/blanc_data_analysis.ipynb [hunk 10]
```diff
@@ -0,0 +1,852 @@
+{
```
Comment (ninalopatina) [inline blanc/results/blanc_data_analysis.ipynb]: I can't review a notebook: I recommend converting much of the notebook to a python script (e.g. via https://medium.com/@researchplex/the-easiest-way-to-convert-jupyter-ipynb-to-python-py-912e39f16917), with its own utils file. The notebook should import functions from utils and only produce figures, with some description of what the figures signify

**Review by ninalopatina — APPROVED** [review]
The code should all be in a directory together, e.g. python. I can review the analysis code in python form, since there are a few changes I would recommend (axes on the last figure, more comments)

File: blanc/README.md (hunk 1 above)
Comment (EvgenyKos) [reply blanc/README.md]: Fixed

Comment (EvgenyKos) [reply blanc/README.md]: Done

File: blanc/README.md (hunk 4 above)
Comment (EvgenyKos) [reply blanc/README.md]: Added into blanc_mt dir.

File: blanc/README.md (hunk 6 above)
Comment (EvgenyKos) [reply blanc/README.md]: Corrected.

**Pull request merged**